### ISSUE TITLE
Use of TypeVar in LHS of member access expression should be an error

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1394]
```python
from typing import Generic, Type, TypeVar
class Base:
    x: Type["Base"]

T = TypeVar("T", bound=Base)

class Foo(Generic[T]):
    def foo(self, a: Type[T.x]) -> T:
        ...
```

The above should generate an error, but it doesn't.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.104, which I just published. It will also be included in the next release of pylance.
